src/interaction/encode.ts

```diff
@@ -50,27 +54,28 @@ export default function(type: Event): void {
             break;
         case Event.InputChange:
             let ch = change.data;
-            tokens.push(observe(ch.target));
+            tokens.push(observe(ch.target as Node));
             tokens.push(ch.value);
             queue(tokens);
             change.reset();
             break;
         case Event.Selection:
             let s = selection.data;
-            tokens.push(observe(s.start));
+            tokens.push(observe(s.start as Node));
             tokens.push(s.startOffset);
-            tokens.push(observe(s.end));
+            tokens.push(observe(s.end as Node));
             tokens.push(s.endOffset);
             queue(tokens);
             selection.reset();
             break;
         case Event.Scroll:
             for (let i = 0; i < scroll.data.length; i++) {
+                if (task.blocking(timer)) { await task.idle(timer); }
```

Comment:
Same comment above as for touch cancel. Why do you have to do this yielding logic only for this event?